sitemap/adder_test.go

```diff
@@ -0,0 +1,101 @@
+package sitemap_test
+
+import (
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/ONSdigital/dp-sitemap/sitemap"
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+func TestAdder(t *testing.T) {
+	Convey("When xml decode returns an error", t, func() {
+		oldSitemap := strings.NewReader("<<<")
+
+		a := &sitemap.DefaultAdder{}
+		filename, err := a.Add(oldSitemap, sitemap.URL{})
+
+		Convey("Adder should return correct error", func() {
+			So(err.Error(), ShouldContainSubstring, "failed to decode old sitemap")
+			So(err.Error(), ShouldContainSubstring, "XML syntax error")
+		})
+		Convey("Temporary sitemap file should be created and then cleaned up", func() {
+			So(filename, ShouldContainSubstring, "sitemap")
+			_, err := os.Stat(filename)
+			So(err.Error(), ShouldContainSubstring, "no such file or directory")
```

Comment:
The err, which gets set in line 25, needs to be initialised otherwise it could lead to a nil pointer reference in line 26.